workers/generic-worker/docker.go

```diff
@@ -12,5 +12,5 @@ func secure(configFile string) {
 }
 
 func MkdirAllTaskUser(dir string, perms os.FileMode) (err error) {
-	return nil
+	return os.MkdirAll(dir, perms)
```

Comment:
Not sure why, for the docker engine, no directory was ever created here.. 🤔

@djmitche - @petemoore is out on PTO now, do you happen to know why this was never implemented like the other generic worker engines?

Reply:
I caught this (_what appears to be a bug_) after writing the purge caches tests.

Reply:
It was because the docker engine used the same user (on the host) for all tasks, like docker-worker did.